infra/token-transfer-server/src/lib/vesting.js

```diff
@@ -1,98 +1,108 @@
-const { GRANT_VEST } = require('../constants/events')
+function toMoment(date) {
+  if (!date) return date
+  if (!moment.isMoment()) return moment(date)
+  return date
+}
 
-async function vestGrant(grant) {
-  const now = grant.now || moment()
+/* Convert the dates of a grant object to moments.
+ *
+ */
+function momentizeGrant(grant) {
+  return {
+    ...grant,
+    start: toMoment(grant.start),
+    end: toMoment(grant.end),
+    cliff: toMoment(grant.cliff),
+    cancelled: toMoment(grant.cancelled)
+  }
+}
 
-  let totalVested = BigNumber(0)
-  const txn = await sequelize.transaction()
-  try {
-    const grantedAt = moment(grant.grantedAt).format('YYYY-MM-DD')
-    let newlyVested = BigNumber(0)
-    // Log an event for each vesting event. This is important in case this is a
-    // catch-up run for this tool. We want a separate log entry for each vesting
-    // event for reporting purposes.
-    for (const vestingEvent of grant.vestingSchedule()) {
-      if (vestingEvent.date.isAfter(now)) {
-        break
-      }
-      totalVested = totalVested.plus(vestingEvent.amount)
-      if (totalVested.isLessThanOrEqualTo(grant.vested)) {
-        continue
-      }
+/**
+ * Returns the vesting schedule for the grant. The vesting schedule is just an
+ * array of integers representing a vested amount. The date/time associated with
+ * each vesting event is not calculated.
+ */
+function vestingEvents(grantObj) {
+  const grant = momentizeGrant(grantObj)
 
-      // Log event and increment vested counter for the grant.
-      const vestingDateStr = vestingEvent.date.format('YYYY-MM-DD')
-      await Event.create({
-        email: grant.email,
-        ip: ip.address(),
-        grantId: grant.id,
-        action: GRANT_VEST,
-        data: JSON.stringify({
-          amount: vestingEvent.amount.toNumber(),
-          vestDate: vestingDateStr
-        })
-      })
-      newlyVested = newlyVested.plus(vestingEvent.amount)
-      if (process.env.NODE_ENV !== 'test') {
-        console.log(
-          chalk`{bold.hex('#26d198') ⬢}  Vested ${vestingEvent.amount.toNumber()} OGN for grant ${grantedAt} to ${
-            grant.email
-          } (effective ${vestingDateStr})`
-        )
-      }
-    }
+  const vestingEventCount = grant.end.diff(grant.start, grant.interval)
+  const vestedPerEvent = BigNumber(grant.amount).div(vestingEventCount)
+  const cliffVestingCount = grant.cliff.diff(grant.start, grant.interval)
 
-    // Sanity checks.
-    grant.vested += newlyVested.toNumber()
-    const calculatedVested = grant.calculateVested()
-    if (grant.vested != calculatedVested) {
-      throw new Error(
-        `vested amount ${grant.vested} != calculated amount ${calculatedVested}`
-      )
-    }
-    if (grant.vested > grant.amount) {
-      throw new Error(
-        `vested ${grant.vested} > grant ${grant.amount} for ${grant.email} ${grantedAt}`
-      )
-    }
+  const cliffVestAmount = Math.floor(vestedPerEvent.times(cliffVestingCount))
```

Comment:
Does Math.floor work on BigNumber objects ?
Maybe safer to use BigNumber rounding. Something like
`vestedPerEvent.times(cliffVestingCount).integerValue(BigNumber.ROUND_FLOOR)`